src/app/container/container.component.html

```diff
@@ -79,16 +79,16 @@ <h3 class="pull-right">
     <h3>
       <div class="pull-right my-entry-button-group">
         <div class="btn-group" role="group">
-          <button id="publishToolButton" type="button" [ngClass]="{'disabled': publishDisable()}" class="btn btn-warning" (click)="publish()" [(ngModel)]="published"
-            [disabled]="!thisisValid" btnCheckbox>
+          <button id="publishToolButton" mat-raised-button color="secondary" (click)="publish()" [(ngModel)]="published"
+            [disabled]="publishDisable()" btnCheckbox [matTooltipShowDelay]="1000" [matTooltipPosition]="'before'" matTooltip="Publish/unpublish the tool">
             {{tool?.is_published ? 'Unpublish' : 'Publish'}}
           </button>
-          <button *ngIf="tool?.is_published" type="button" [disabled]="tool?.is_published" class="btn btn-danger">
+          <button *ngIf="tool?.is_published" mat-raised-button color="warn" [disabled]="tool?.is_published" [matTooltipShowDelay]="1000" [matTooltipPosition]="'before'" matTooltip="Delete the tool">
             Delete
           </button>
-          <app-deregister-modal *ngIf="!tool?.is_published"></app-deregister-modal>
+          <app-deregister-modal *ngIf="!tool?.is_published" [refreshMessage]="refreshMessage"></app-deregister-modal>
         </div>
-        <button type="button" class="btn btn-primary" (click)="refresh()" [disabled]="refreshMessage !== null">
+        <button mat-raised-button color="primary" (click)="refresh()" [disabled]="refreshMessage !== null && refreshMessage !== undefined" [matTooltipShowDelay]="1000" [matTooltipPosition]="'before'" matTooltip="Refresh the tool">
```

Comment:
You could probably simplify the disabled to `!refreshMessage`. Although if you did that, it would also disable on an empty string -- would you want that?

Reply:
I think that should be fine.

Reply:
When I replace the existing refreshMessage !== null && refreshMessage !== undefined with !refreshMessage it doesn't seem to work. I'm thinking that the empty string may be important, so I am going to leave it as is for now.

Reply:
umm...should be replacing it with `refreshMessage`

Reply:
Oops, right.